src/bech32.c

```diff
@@ -21,36 +21,44 @@
 #include "internal.h"
 #include <stdlib.h>
 #include <stdint.h>
+#include <assert.h>
 #include <include/wally_address.h>
 #include <include/wally_script.h>
 #include <include/wally_crypto.h>
 #include "script.h"
+#include "bech32.h"
 
 
 static uint32_t bech32_polymod_step(uint32_t pre) {
     uint8_t b = pre >> 25;
     return ((pre & 0x1FFFFFF) << 5) ^
-           (-((b >> 0) & 1) & 0x3b6a57b2UL) ^
-           (-((b >> 1) & 1) & 0x26508e6dUL) ^
-           (-((b >> 2) & 1) & 0x1ea119faUL) ^
-           (-((b >> 3) & 1) & 0x3d4233ddUL) ^
-           (-((b >> 4) & 1) & 0x2a1462b3UL);
+        (-((b >> 0) & 1) & 0x3b6a57b2UL) ^
+        (-((b >> 1) & 1) & 0x26508e6dUL) ^
+        (-((b >> 2) & 1) & 0x1ea119faUL) ^
+        (-((b >> 3) & 1) & 0x3d4233ddUL) ^
+        (-((b >> 4) & 1) & 0x2a1462b3UL);
 }
 
-static const char *charset = "qpzry9x8gf2tvdw0s3jn54khce6mua7l";
+static uint32_t bech32_final_constant(bech32_encoding enc) {
+    if (enc == BECH32_ENCODING_BECH32) return 1;
+    if (enc == BECH32_ENCODING_BECH32M) return 0x2bc830a3;
+    assert(0);
```

Comment:
we can't use assert for error handling in wally, only return error codes to the caller.

Reply:
This code path should never hit. So, having an assert actually helps with the debugging if somehow it occurs in the execution. But for now, to match with wally's error handling, I am just returning a placeholder value.